src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -2090,11 +2090,17 @@ public IAsyncResult BeginConnect(EndPoint remoteEP, AsyncCallback? callback, obj
 
         private bool CanUseConnectEx(EndPoint remoteEP)
         {
-            return (_socketType == SocketType.Stream) &&
-                (_rightEndPoint != null || remoteEP.GetType() == typeof(IPEndPoint));
-        }
+            Debug.Assert(remoteEP.GetType() != typeof(DnsEndPoint));
 
+            // ConnectEx supports connection-oriented sockets.
+            // The socket must be bound before calling ConnectEx.
+            //     In case of IPEndPoint, the Socket will be bound using WildcardBindForConnectIfNecessary.
+            // Unix sockets are not supported by ConnectEx.
 
+            return (_socketType == SocketType.Stream) &&
+                   (_rightEndPoint != null || remoteEP.GetType() == typeof(IPEndPoint)) &&
+                   (remoteEP.AddressFamily != AddressFamily.Unix);
```

Comment:
I wonder if @stephentoub's https://github.com/dotnet/runtime/pull/33674#discussion_r395245094 also applies to `AddressFamily` ? Wouldn't it be more future proof to use the condition `(remoteEP.AddressFamily == AddressFamily.InterNetwork || remoteEP.AddressFamily == AddressFamily.InterNetworkV6)` instead?

Reply:
I have no preference for `remoteEP.GetType() == typeof(IPEndPoint)` vs `(remoteEP.AddressFamily == AddressFamily.InterNetwork || remoteEP.AddressFamily == AddressFamily.InterNetworkV6)`, so I kept the check as it was.